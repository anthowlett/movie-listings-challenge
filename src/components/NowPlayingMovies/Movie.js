import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { breakpoints } from '../../utils';
import Image from './Image';

// Styles
const Wrapper = styled.div`
  flex-basis: 100%;
  
  ${breakpoints.mini(`
      flex-basis: 50%;
  `)};
  ${breakpoints.small(`
      flex-basis: 33.333%;
  `)};
  ${breakpoints.medium(`
      flex-basis: 25%;
  `)};
`;

const Details = styled.div`
  padding: 5px 15px;
`;

const Movie = ({
  data,
  genres
}) => (
	<Wrapper>
		<Image name={data.poster_path} alt={data.title} />
		<Details>
			<h2>{data.title}</h2>
			<div>
				<p>Rated <strong>{data.vote_average}</strong> / 10</p>
				<p>Popularity <strong>{Math.round(data.popularity)}</strong></p>
        <ul>
          {
            genres
              .filter(genre => data.genre_ids.includes(genre.id))
              .map((genre) => (
                <li key={genre.id}>{genre.name}</li>
              ))
          }
        </ul>
			</div>
		</Details>
	</Wrapper>
);

Movie.propTypes = {
  genres: PropTypes.array.isRequired,
  data: PropTypes.object.isRequired
};

export default Movie;