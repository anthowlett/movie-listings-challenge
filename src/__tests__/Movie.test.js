import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import Movie from '../components/NowPlayingMovies/Movie';
import '../setupTests';

const mockStore = configureMockStore();
const store = mockStore({});

describe('NowPlayingMovies', () => {
  
  describe('<Movie />', () => {
    const genres = [
      {id: 28, name: "Action"},
      {id: 12, name: "Adventure"},
      {id: 16, name: "Animation"}
    ]; 
    const item = {
      id: 450465,
      title: 'Aquaman',
      genre_ids: [28, 12, 16],
      popularity: 200
    };
    const component = shallow(
      <Movie store={store} key={item.id} data={item} genres={genres} />
    )

    it('should render', () => {
      expect(component.debug())
        .toMatchSnapshot();
    });
  });

});