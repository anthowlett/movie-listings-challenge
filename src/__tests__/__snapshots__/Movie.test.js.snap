// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NowPlayingMovies <Movie /> should render 1`] = `
"<styled.div>
  <Connect(Image) name={[undefined]} alt=\\"Aquaman\\" />
  <styled.div>
    <h2>
      Aquaman
    </h2>
    <div>
      <p>
        Rated 
        <strong />
         / 10
      </p>
      <p>
        Popularity 
        <strong>
          200
        </strong>
      </p>
      <ul>
        <li>
          Action
        </li>
        <li>
          Adventure
        </li>
        <li>
          Animation
        </li>
      </ul>
    </div>
  </styled.div>
</styled.div>"
`;
